install.sh

```diff
@@ -5,6 +5,15 @@ if [[ -n "$MSYSTEM" ]]; then
   exit 1
 fi
 
+if [[ -d ".git" ]]; then
+  if [[ $(git branch | sed -n '/\* /s///p') == "master" ]]; then
+    if [[ $(git rev-parse HEAD) != $(git ls-remote $(git rev-parse --abbrev-ref @{u} | sed 's/\// /g') | cut -f1) ]]; then
```

Comment:
Some comments here explaining the logic would be very beneficial.

I also think we should have a flag (CLI or env variable) to bypass this logic as I can easily see this becoming a nuisance for people maintaining their custom versions.